cirq-pasqal/cirq_pasqal/pasqal_sampler.py

```diff
@@ -21,15 +21,20 @@
 
 
 class PasqalSampler(cirq.work.Sampler):
-    def __init__(self, remote_host: str, access_token: str = '') -> None:
+    def __init__(
+        self, remote_host: str, access_token: str = '', device: cirq_pasqal.PasqalDevice = None
```

Comment:
We do require a `PasqalDevice` to be specified. If that's not going to be done in `Circuit` anymore, this shouldn't be optional.

Reply:
>We do require a PasqalDevice to be specified.

I can add the assertion checks at init to make sure it's a pasqal device.

> If that's not going to be done in Circuit anymore, device shouldn't be optional and it must be a PasqalDevice instance.

This would make this a breaking change. If this PR were to be merged, both functionality would still work. Until the deprecation warning is up a user could still attach a device to a circuit or construct a sampler with a device (sampler device taking precedence here). Would you prefer we do a breaking style change and make this argument required ?

Reply:
I see what you mean. While the option to attach a device to a circuit is still there, this doesn't need to be required. What if you keep it `Optional` for now, but demand that there is a `PasqalDevice` associated? If a `PasqalDevice` is attached to the circuit, this doesn't have to be specified. Otherwise, it is required.  Later on, when the option to have a device attached to a circuit is gone, this is turned into a required argument.

Reply:
Did you consider this option?

Reply:
I believe so. To me this sounds like what is currently implemented in the code. Re-reading it now I'm having trouble figuring out what we're missing, would you mind adding a little snippet here in python or pseudocode of what changes we should consider to help me understand ?

Reply:
Sure, see my comment on lines 136 to 142.